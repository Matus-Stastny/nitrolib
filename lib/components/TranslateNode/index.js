"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _identity2 = _interopRequireDefault(require("ramda/src/identity"));

var React = _interopRequireWildcard(require("react"));

var _context = require("../../services/intl/context");

/* eslint-disable react/no-danger, react/no-array-index-key */
var TranslateNode = function TranslateNode(_ref) {
  var t = _ref.t,
      values = _ref.values,
      transform = _ref.transform;
  return React.createElement(_context.Consumer, null, function (intl) {
    return transform(intl.translate(t)).split("__").filter(Boolean).map(function (word, i) {
      return values[word] ? React.createElement("span", {
        key: i
      }, values[word]) : React.createElement("span", {
        key: i,
        dangerouslySetInnerHTML: {
          __html: word
        }
      });
    });
  });
};

TranslateNode.defaultProps = {
  transform: _identity2.default
};
var _default = TranslateNode;
exports.default = _default;