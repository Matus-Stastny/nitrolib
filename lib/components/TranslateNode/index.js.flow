// @flow strict
import * as React from "react";

type Props = {|
  t: string,
  values: { [key: string]: React.Node },
  // defaulted
  transform?: (value: string) => string, // identity
|};

declare export default React.ComponentType<Props>;
