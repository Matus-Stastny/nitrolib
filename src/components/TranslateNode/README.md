# TranslateNode

The same as `Translate`, except values are `React.Node`. Useful when you need to inject elements with event handlers!

> The text output can be transformed using the `transform` function. You can use arrow functions - the component does not implement pure render.

Context needs:
* **intl**

Example:
```js
const MyComponent = () => (
  <TranslateNode
    t="Click this: __x__"
    values={
      { x: <button onClick={() => alert("Clicked")}>Yo</button> }
    }
  />
);
```
