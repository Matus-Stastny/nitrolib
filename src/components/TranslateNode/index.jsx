// @flow strict
import * as React from "react";

import { Consumer } from "../../services/intl/context";

type Props = {
  t: string,
  values: { [key: string]: React.Node },
};

const TranslateNode = ({ t, values }: Props) => (
  <Consumer>
    {intl => (
      <>
        {Object.keys(values).reduce(
          // $FlowExpected: we're transforming string[] to React.Node[]
          (words, key) => words.map(word => (word === key ? values[key] : word)),
          intl.translate(t).split("__"),
        )}
      </>
    )}
  </Consumer>
);

export default TranslateNode;
