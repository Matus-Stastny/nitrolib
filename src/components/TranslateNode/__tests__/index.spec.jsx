// @flow strict
import * as React from "react";
import { shallow } from "enzyme";

import TranslateNode from "..";

describe("#TranslateNode", () => {
  test("string", () => {
    const wrapper = shallow(
      <TranslateNode t="lol __x__ kek bur" values={{ x: <span>lmao</span> }} />,
    );

    expect(wrapper).toMatchSnapshot();
  });
});
