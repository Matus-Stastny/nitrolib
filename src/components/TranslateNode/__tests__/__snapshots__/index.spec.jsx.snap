// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#TranslateNode string 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
